sdk/tests/conformance/canvas/rapid-resizing.html

```diff
@@ -58,80 +59,97 @@
 description("Verifies that rapidly resizing the canvas works correctly.");
 
 debug("");
-debug("Regression test for Chromium <a href='http://crbug.com/299371'>Issue 299371</a>");
+debug("Regression test for Chromium <a href='http://crbug.com/299371'>Issue 299371</a> and <a href='http://crbug.com/557848'>Issue 557848</a>");
 debug("");
 
 var err;
 var wtu = WebGLTestUtils;
-var canvas = document.getElementById("canvas");
+var canvas;
 var largeSize = 256;
 var smallSize = 128;
-var currentSize = largeSize;
-canvas.width = largeSize;
-canvas.height = largeSize;
-var gl = wtu.create3DContext(canvas);
+var currentSize;
+var gl;
+var program;
 var numFrames = 0;
+var testNumber = 0;
 
-if (!gl) {
-  testFailed("context does not exist");
-} else {
-  testPassed("context exists");
-
-  gl.clearColor(0, 0, 0, 1);
-
-  var program = wtu.setupProgram(gl, ["vshader", "fshader"], ["position"]);
-  shouldBeNonNull("program");
-
-  // Prepare to draw quads
-  var quadSize = 0.1;
-
-  var vertexBuffer = gl.createBuffer();
-  gl.bindBuffer(gl.ARRAY_BUFFER, vertexBuffer);
-  gl.bufferData(gl.ARRAY_BUFFER, new Float32Array([
-    // Lower left
-    -1, -1 + quadSize,
-    -1, -1,
-    -1 + quadSize, -1,
-    -1 + quadSize, -1 + quadSize,
-
-    // Lower right
-    1 - quadSize, -1 + quadSize,
-    1 - quadSize, -1,
-    1, -1,
-    1, -1 + quadSize,
-
-    // Upper right
-    1 - quadSize, 1,
-    1 - quadSize, 1 - quadSize,
-    1, 1 - quadSize,
-    1, 1,
-
-    // Upper left
-    -1, 1,
-    -1, 1 - quadSize,
-    -1 + quadSize, 1 - quadSize,
-    -1 + quadSize, 1
-  ]), gl.STATIC_DRAW);
-  gl.enableVertexAttribArray(0);
-  gl.vertexAttribPointer(0, 2, gl.FLOAT, false, 0, 0);
-
-  var indexBuffer = gl.createBuffer();
-  gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, indexBuffer);
-  gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, new Uint16Array([
-    0, 1, 2,
-    0, 2, 3,
-
-    4, 5, 6,
-    4, 6, 7,
-
-    8, 9, 10,
-    8, 10, 11,
-
-    12, 13, 14,
-    12, 14, 15
-  ]), gl.STATIC_DRAW);
-
-  wtu.requestAnimFrame(render);
+function nextTest() {
```

Comment:
This is largely a refactoring to let the main loop of the test run twice, once with preserveDrawingBuffer=true and once with it false. Unfortunately it's not easy to see that in the code review; the refactoring is going to be hard to review anyway, so I think we should just go ahead with the change.